### PR TITLE
Fix amici error output, fixes #427

### LINKED ISSUE #427: Aggregated objective fails on integration errors
> [Warning] AMICI:simulation: AMICI simulation failed:
Steady state computation failed. First run of Newton solver failed: No convergence was achieved. Simulation to steady state failed. Second run of Newton solver failed: No convergence was achieved.
Error occured in:
0       0x7f1dc1a73a73 amici::AmiException::AmiException(char const*, ...) + 179
1       0x7f1dc1a7ca30 amici::SteadystateProblem::handleSteadyStateFailure(amici::Solver const*, amici::Model*) + 352
2       0x7f1dc1a7d98c /home/ff72/melanoma/erk_pulses_calibration/venv/src/amici/python/sdist/amici/_amici.cpython-37m-x86_64-linux-gnu.so(+0x1d198c) [0x7f1dc1a7d98c]
3       0x7f1dc1a7da0d amici::SteadystateProblem::workSteadyStateProblem(amici::Solver*, amici::Model*, int) + 125
4       0x7f1dc1a2
[Warning] AMICI:simulation: AMICI simulation failed:
Steady state computation failed. First run of Newton solver failed: No convergence was achieved. Simulation to steady state failed. Second run of Newton solver failed: No convergence was achieved.
Error occured in:
0       0x7f1dc1a73a73 amici::AmiException::AmiException(char const*, ...) + 179
1       0x7f1dc1a7ca30 amici::SteadystateProblem::handleSteadyStateFailure(amici::Solver const*, amici::Model*) + 352
2       0x7f1dc1a7d98c /home/ff72/melanoma/erk_pulses_calibration/venv/src/amici/python/sdist/amici/_amici.cpython-37m-x86_64-linux-gnu.so(+0x1d198c) [0x7f1dc1a7d98c]
3       0x7f1dc1a7da0d amici::SteadystateProblem::workSteadyStateProblem(amici::Solver*, amici::Model*, int) + 125
4       0x7f1dc1a2
[Warning] AMICI:simulation: AMICI simulation failed:
Steady state computation failed. First run of Newton solver failed: No convergence was achieved. Simulation to steady state failed. Second run of Newton solver failed: No convergence was achieved.
Error occured in:
0       0x7f1dc1a73a73 amici::AmiException::AmiException(char const*, ...) + 179
1       0x7f1dc1a7ca30 amici::SteadystateProblem::handleSteadyStateFailure(amici::Solver const*, amici::Model*) + 352
2       0x7f1dc1a7d98c /home/ff72/melanoma/erk_pulses_calibration/venv/src/amici/python/sdist/amici/_amici.cpython-37m-x86_64-linux-gnu.so(+0x1d198c) [0x7f1dc1a7d98c]
3       0x7f1dc1a7da0d amici::SteadystateProblem::workSteadyStateProblem(amici::Solver*, amici::Model*, int) + 125
4       0x7f1dc1a2
Traceback (most recent call last):
  File "parameter_estimation.py", line 55, in <module>
    history_options=history_options
  File "/home/ff72/melanoma/erk_pulses_calibration/venv/lib/python3.7/site-packages/pypesto/optimize/optimize.py", line 82, in minimize
    problem=problem, options=options)
  File "/home/ff72/melanoma/erk_pulses_calibration/venv/lib/python3.7/site-packages/pypesto/startpoint/util.py", line 67, in assign_startpoints
    method=startpoint_method
  File "/home/ff72/melanoma/erk_pulses_calibration/venv/lib/python3.7/site-packages/pypesto/startpoint/util.py", line 91, in resample_startpoints
    fvals[j] = problem.objective(startpoint)
  File "/home/ff72/melanoma/erk_pulses_calibration/venv/lib/python3.7/site-packages/pypesto/objective/base.py", line 145, in __call__
    result = self.call_unprocessed(x_full, sensi_orders, mode)
  File "/home/ff72/melanoma/erk_pulses_calibration/venv/lib/python3.7/site-packages/pypesto/objective/aggregated.py", line 74, in call_unprocessed
    for objective in self._objectives
  File "/home/ff72/melanoma/erk_pulses_calibration/venv/lib/python3.7/site-packages/pypesto/objective/aggregated.py", line 100, in aggregate_results
    for key in [FVAL, CHI2, SCHI2, GRAD, HESS, HESSP]
  File "/home/ff72/melanoma/erk_pulses_calibration/venv/lib/python3.7/site-packages/pypesto/objective/aggregated.py", line 101, in <dictcomp>
    if rvals[0].get(key, None) is not None
  File "/home/ff72/melanoma/erk_pulses_calibration/venv/lib/python3.7/site-packages/pypesto/objective/aggregated.py", line 99, in <genexpr>
    key: sum(rval[key] for rval in rvals)
KeyError: 'grad'

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/objective/amici_util.py  4
         

Complexity decreasing per file
==============================
+ pypesto/objective/amici_calculator.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5835389&bid=19108848)

File: pypesto/objective/amici_util.py [hunk 1]
```diff
@@ -237,11 +241,38 @@ def get_error_output(
...
-        GRAD: np.nan * np.ones(dim),
-        HESS: np.nan * np.ones([dim, dim]),
-        RES:  np.nan * np.ones(n_res),
-        SRES: np.nan * np.ones([n_res, dim]),
+    nllh = np.inf
+    snllh = None
+    s2nllh = None
+    if mode == MODE_FUN and sensi_order > 0:
+        snllh = np.nan * np.ones(dim)
+        s2nllh = np.nan * np.ones([dim, dim])
```
Comment (yannikschaelte) [inline pypesto/objective/amici_util.py:249]: I think for compliance with the non-error output mode, one should check for sensi_order > 1?

Comment (FFroehlich) [reply pypesto/objective/amici_util.py:249]: I copy pasted the code from amici-calculator, might be that they are initialized but not modified. Will look into in more detail.

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/objective/amici_util.py  4
         

Complexity decreasing per file
==============================
+ pypesto/objective/amici_calculator.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5835389&bid=19108848)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/objective/amici_util.py  2
         

Complexity decreasing per file
==============================
+ pypesto/objective/amici_calculator.py  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5835389&bid=19108848)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/objective/amici_util.py  2
         

Complexity decreasing per file
==============================
+ pypesto/objective/amici_calculator.py  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5835389&bid=19108848)

**Pull request merged**